src/db/drivers/mysql.ts

```diff
@@ -0,0 +1,219 @@
+import { Sequelize, DataTypes, Model, BuildOptions } from "sequelize";
+import DatabaseDriver from "./driver";
+import User from "./../../models/user";
+import StudyGroup from "./../../models/group";
+
+
+// TypeScript has trouble generating a Model definition with sequelize.define,
+// so we need to define some types to help out with that
+interface SequelizeUserModel extends Model {
+  readonly id: number;
+  readonly username: string;
+}
+
+type StaticSequelizeUserModel = typeof Model & {
+  new(values?: object, options?: BuildOptions): SequelizeUserModel;
+}
+
+interface SequelizeStudyGroupModel extends Model {
+  readonly id: number;
+  readonly groupName: string;
+}
+
+type StaticSequelizeStudyGroupModel = typeof Model & {
+  new(values?: object, options?: BuildOptions): SequelizeStudyGroupModel;
+}
+
+/**
+ * MySQL implementation of a database driver.
+ */
+export default class MySqlDriver extends DatabaseDriver {
+  private sequelize: Sequelize;
+  private sequelizeUserModel: StaticSequelizeUserModel;
+  private sequelizeStudyGroupModel: StaticSequelizeStudyGroupModel;
+
+  /**
+   * Constructor of the MySQL adapter.
+   * 
+   * @param databaseName name of the MySQL database to use
+   * @param username username used to authenticate against the database
+   * @param password password used to authenticate against the database
+   * @param hostName host of the MySQL database
+   */
+  constructor(databaseName: string, username: string, password: string, host: string) {
+    super();
+
+    this.sequelize = new Sequelize(
+      databaseName,
+      username,
+      password,
+      {
+        host: host,
+        dialect: "mysql"
+      }
+    );
+
+    // define our user model for sequelize
+    this.sequelizeUserModel = this.sequelize.define("User", {
+      id: {
+        primaryKey: true,
+        type: DataTypes.INTEGER.UNSIGNED,

```

Comment:
Should allowNull be false for the id values?

Reply:
Done